src/effects/VST3/VST3Effect.cpp

```diff
@@ -687,6 +758,10 @@ bool VST3Effect::RealtimeAddProcessor(
       auto effect = std::make_unique<VST3Effect>(*this);
       effect->mSetup.processMode = Vst::kRealtime;
       effect->mSetup.sampleRate = sampleRate;
+      //IAudioProcessor should be cofigured for a different processing mode
+      if(!SetupProcessing(*effect->mEffectComponent.get(), effect->mSetup))
+         return false;
```

Comment:
In this place, you are not careful, as with other calls to `SetupProcessing` to avoid side effects on mSetup in case of failure.  Should that change?

There are also two calls in constructors of VST3Effect but I think I understand that it is not necessary there.

Reply:
If realtime effect instance cannot accept configuration than it's completely discarded and `RealtimeAddProcessor` reports failure